packages/injector/src/module.ts

```diff
@@ -180,8 +180,9 @@ export function findModuleForConfig(config: ClassType, modules: InjectorModule[]
 
 export type ExportType = Token | InjectorModule;
 
-export function isProvided(providers: ProviderWithScope[], token: any): boolean {
-    return providers.find(v => !(v instanceof TagProvider) ? token === (isClass(v) ? v : v.provide) : false) !== undefined;
+
+export function isProvided<T>(providers: ProviderWithScope[], token: Token<T>): boolean {
+    return providers.some(v => !(v instanceof TagProvider) ? token === (isClass(v) ? v : v.provide) : false);
```

Comment:
In the past it was probably enough to match for classType being equal. Since Module.isProvided also allows to use `Type` we need to be a bit more fancy here. We need the new function `getContainerToken` which returns a comparable value and apply it to `token` and each provider in the ProviderWithScope provider.
Something like:

```typescript
providers.some(v => getContainerToken(v.provide) === getContainerToken(token));
```

this does obviously not check for any structural compatibility or nominal inheritance.

Reply:
See https://github.com/deepkit/deepkit-framework/blob/21822f905edf4fc4fdef4d19371b20fb3c5f450b/packages/injector/src/injector.ts#L187-L205

Reply:
@marcj done